src/App/Pages/Accounts/LoginSsoPage.xaml.cs

```diff
@@ -0,0 +1,105 @@
+
+        public LoginSsoPage(AppOptions appOptions = null)
+        {
+            _storageService = ServiceContainer.Resolve<IStorageService>("storageService");
+            _messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
+            _vaultTimeoutService = ServiceContainer.Resolve<IVaultTimeoutService>("vaultTimeoutService");
+            _messagingService.Send("showStatusBar", true);
+            _appOptions = appOptions;
+            InitializeComponent();
+            _vm = BindingContext as LoginSsoPageViewModel;
+            _vm.Page = this;
+            _vm.StartTwoFactorAction = () => Device.BeginInvokeOnMainThread(async () => await StartTwoFactorAsync());
+            _vm.StartSetPasswordAction = () =>
+                Device.BeginInvokeOnMainThread(async () => await StartSetPasswordAsync());
+            _vm.LoggedInSsoAction = () => Device.BeginInvokeOnMainThread(async () => await LoggedInSsoAsync());
+            _vm.CloseAction = async () =>
+            {
+                _messagingService.Send("showStatusBar", false);
+                await Navigation.PopModalAsync();
+            };
+            if (Device.RuntimePlatform == Device.Android)
+            {
+                ToolbarItems.RemoveAt(0);
+            }
+        }
+
+        protected override async void OnAppearing()
+        {
+            base.OnAppearing();
+            await _vm.InitAsync();
+            if (string.IsNullOrWhiteSpace(_vm.OrgIdentifier))
+            {
+                RequestFocus(_orgIdentifier);
+            }
+        }
+
+        private async void LogIn_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.LogInAsync();
+            }
+        }
+
+        private async void Close_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                _vm.CloseAction();
+            }
+        }
+
+        private async Task StartTwoFactorAsync()
+        {
+            var page = new TwoFactorPage(true);
+            await Navigation.PushModalAsync(new NavigationPage(page));
+        }
+
+        private async Task StartSetPasswordAsync()
+        {
+            var page = new SetPasswordPage(_appOptions);
+            await Navigation.PushModalAsync(new NavigationPage(page));
+        }
+
+        private async Task LoggedInSsoAsync()
+        {
+            if (_appOptions != null)
+            {
+                if (_appOptions.FromAutofillFramework && _appOptions.SaveType.HasValue)
+                {
+                    Application.Current.MainPage = new NavigationPage(new AddEditPage(appOptions: _appOptions));
+                    return;
+                }
+                if (_appOptions.Uri != null)
+                {
+                    Application.Current.MainPage = new NavigationPage(new AutofillCiphersPage(_appOptions));
+                    return;
+                }
+            }
```

Comment:
Can we actually navigate to these on SSO login?

Reply:
It does now :)